packages/server/src/fhir/repo.ts

```diff
@@ -602,13 +602,15 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
   }
 
   private async loadProfile(url: string): Promise<StructureDefinition | undefined> {
-    const projectId = this.context.project;
-
-    if (projectId) {
-      // Try retrieving from cache
-      const cachedProfile = await getProfileCacheEntry(projectId, url);
-      if (cachedProfile) {
-        return cachedProfile.resource;
+    const projectIds = this.context.projects;
+
+    if (projectIds?.length) {
+      for (const projectId of projectIds) {
```

Comment:
Low priority, but this could be converted to a Redis `mget` for a single parallelized operation.